test/browser/component.js

```diff
@@ -826,3 +826,20 @@ describe(`Lifecycle Helpers`, function () {
     });
   });
 });
+
+describe(`Required Attrs Components`, function () {
```

Comment:
I'm not too sure what this test tests. I know @tdumitrescu has stronger opinions on test structure. Please wait on his 👍 /  ✅

Reply:
structurally it's fine, though think about the English statement constructed by the nested levels: "Required Attrs Components renders". Maybe a `context()` would be better, to produce something like "Component / with required attrs / renders successfully when the attrs are provided" and "Component / with required attrs / throws an error when the attrs are not provided". No biggie either way, though it would be good to have the non-happy-path tested in this module.

Reply:
I was struggling to make a non-happy path work. I was wrapping the `.appendChild` with an `expect(...).to.throw` but it would never trip. Do you have any suggestions?

Reply:
yeah presumably in the real browser test the error would be getting thrown asynchronously on connect rather than directly in the appendChild stack, right? if you remove the expect.to.throw does it error out and fail the test at least?

Reply:
it always throws and always fails the test due to the async nature. I was unable to successfully catch it to make an assertion that it failed for the reason I expected.